### PR TITLE
Add app registration to payment management service

### PR DESCRIPTION
**Pull request by steve-dickinson**
https://eaflood.atlassian.net/browse/SFI-1422

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,13 +4,39 @@ FFC payment services management
...
 - Docker
 - Docker Compose
 
 Optional:
 - Kubernetes
 - Helm
 
+## Azure App Registration
+
+`ffc-pay-web` has been integrated into Azure App Registration using the msal-node npm packackage (https://www.npmjs.com/package/@azure/msal-node)
```
Comment (johnwatson484) [inline README.md:17]: Typo: `packackage`

I'd also suggest making the link a markdown link.  eg: `[npm package](https://www.npmjs.com/package/@azure/msal-node)`

File: README.md [hunk 2]
```diff
@@ -4,13 +4,39 @@ FFC payment services management
...
+
+For `ffc-pay-web` to authenticate it needs to be registered with Azure App Registration (https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app)
```
Comment (johnwatson484) [inline README.md:19]: As above I'd make this a markdown link with syntax [ ] ( )

File: README.md [hunk 3]
```diff
@@ -4,13 +4,39 @@ FFC payment services management
...
+The following environment varibles need to be set:
+
+- AZURE_CLIENT_ID
+- AZURE_TENANT_ID
+- AZURE_CLIENT_SECRET
+- REDIRECT_URL
+
+These can be retrieved from the App Restistration overview blade.
+
+The following roles need setting up (https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-add-app-roles-in-azure-ad-apps)
```
Comment (johnwatson484) [inline README.md:30]: As above I'd make this a markdown link with syntax [ ] ( )

File: app/permissions.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+const permissions = {
+  viewPaymentHolds: 'Payments.View.Holds',
+  viewPaymentScheme: 'Payments.View.Scheme',
+  updatePaymentScheme: 'Payments.Update.Scheme',
+  addPaymentHold: 'Payments.Add.Hold',
+  removePaymentHold: 'Payments.Remove.Hold'
```
Comment (johnwatson484) [inline app/permissions.js:6]: Not really in the scope of this PR but we will need to refine these roles with the business.

File: app/routes/authenticate.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  handler: async (request, h) => {
+    try {
+      await azureAuth.authenticate(request.query.code, request.cookieAuth)
+      return h.redirect('/')
+    } catch (err) {
+      console.log('Error authenticating')
+      console.log(JSON.stringify(err))
+    }
+
+    return h.view('500').code(500)
```
Comment (johnwatson484) [inline app/routes/authenticate.js:18]: Would this be an error in the authentication process and there 500 is fine.  Or would failed authentication throw an error in which case 401 or 403 would be a better fit?

File: app/routes/cookies.js [hunk 6]
```diff
@@ -6,13 +6,17 @@ const ViewModel = require('./models/cookies-policy')
...
     return h.view('cookies/cookie-policy', new ViewModel(request.state[cookiesPolicy], request.query.updated))
   }
 }, {
   method: 'POST',
   path: '/cookies',
   options: {
+    auth: false,
     plugins: {
       crumb: false
     },
```
Comment (johnwatson484) [inline app/routes/cookies.js:22]: Not related to this PR, but there's no value in a cookie policy in this service.  We probably can strip it all out as part of the ticket we have to make the branding suitable for internal applications.

File: README.md [hunk 7]
```diff
@@ -4,13 +4,39 @@ FFC payment services management
...
+`ffc-pay-web` has been integrated into Azure App Registration using the msal-node npm packackage (https://www.npmjs.com/package/@azure/msal-node)
+
+For `ffc-pay-web` to authenticate it needs to be registered with Azure App Registration (https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app)
+
+The following environment varibles need to be set:
+
+- AZURE_CLIENT_ID
+- AZURE_TENANT_ID
+- AZURE_CLIENT_SECRET
+- REDIRECT_URL
```
Comment (johnwatson484) [inline README.md:26]: Config specifies a default value for `REDIRECT_URL` so this one doesn't need to be set and can be dropped.

Though see my comment about the default values not working

**Review by johnwatson484 — COMMENTED** [review]
Added a few initial comments.  Mostly questions and future considerations.

**Review by johnwatson484 — COMMENTED** [review]
Could we avoid the need for a login button?

ie user access any url, if they're not authenticated we trigger the process.  Rather than giving them a login button to click.  The less button clicks the better

File: app/permissions.js (hunk 4 above)
Comment (simonnineteen90) [reply app/permissions.js:6]: Added these role for testing only

File: app/routes/authenticate.js (hunk 5 above)
Comment (simonnineteen90) [reply app/routes/authenticate.js:18]: This is for an error in the auth process so think code 500 is correct

**Comment by simonnineteen90** [comment]
> 
Updated to remove login view and login button

File: README.md (hunk 7 above)
Comment (simonnineteen90) [reply README.md:26]: removed

File: app/routes/cookies.js (hunk 6 above)
Comment (simonnineteen90) [reply app/routes/cookies.js:22]: Have gone through and removed the cookie policy code

**Review by johnwatson484 — APPROVED** [review]

**Pull request merged**